### PR TITLE
Adds the unit tests to the CI

### PR DESCRIPTION
**Pull request by ike709**
Willox added unit tests, this should run them on every PR.

### CODE REVIEW

**Comment by ike709** [comment]
Unit tests... currently don't work? @willox

**Comment by willox** [comment]
The error code from the tests match dotnet's `CoreHostLibMissingFailure` error code, which suggests that the compiler launched through the test isn't able to find the runtime installed by `actions/setup-dotnet`?

https://github.com/wixoaGit/OpenDream/blob/aedc578e8bbef62a31698c80313d28dd8b9a2969/OpenDreamRuntime.Tests/UnitTest1.cs#L55-L66

I have tested the tests on a fresh install of Windows (with dotnet installed) successfully, maybe it'll fail if I use a zip version of dotnet instead of the installed version?

**Comment by willox** [comment]
Even with dotnet in a zip rather than being installed, `../dotnet/dotnet.exe test` successfully runs DMCompile.exe on a fresh Windows install for me...

**Comment by ike709** [comment]
Yeah I have no ideas

**Comment by ZeWaka** [comment]
I'll take a look at this when I get a chance.

File: .github/workflows/compiler-test.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ jobs:
       run: dotnet build main/OpenDream.sln --configuration Release --no-restore /m
     - name: Compile TestGame
       run: main\DMCompiler\bin\Release\net5.0\DMCompiler.exe main\TestGame\environment.dme
+    - name: Run Unit Tests
+      run: dotnet test main/OpenDream.sln -c Release
```
Comment (ike709) [inline .github/workflows/compiler-test.yml:28]:
```suggestion
      run: dotnet test --no-build main/OpenDream.sln
```

File: .github/workflows/compiler-test.yml [hunk 2]
```diff
@@ -24,6 +24,8 @@ jobs:
       run: dotnet build main/OpenDream.sln --configuration Release --no-restore /m
     - name: Compile TestGame
       run: main\DMCompiler\bin\Release\net5.0\DMCompiler.exe main\TestGame\environment.dme
+    - name: Run Unit Tests
+      run: dotnet test --no-build main/OpenDream.sln
```
Comment (ike709) [inline .github/workflows/compiler-test.yml:28]:
```suggestion
      run: dotnet test --no-build main\OpenDreamRuntime.Tests\OpenDreamRuntime.Tests.csproj
```

File: .github/workflows/compiler-test.yml [hunk 3]
```diff
@@ -24,6 +24,8 @@ jobs:
       run: dotnet build main/OpenDream.sln --configuration Release --no-restore /m
     - name: Compile TestGame
       run: main\DMCompiler\bin\Release\net5.0\DMCompiler.exe main\TestGame\environment.dme
+    - name: Run Unit Tests
+      run: dotnet test --no-build main\OpenDreamRuntime.Tests\OpenDreamRuntime.Tests.csproj
```
Comment (ike709) [inline .github/workflows/compiler-test.yml:28]:
```suggestion
      run: dotnet test --no-build main\OpenDreamRuntime.Tests\OpenDreamRuntime.Tests.csproj --configuration Release
```

**Comment by ike709** [comment]
See https://github.com/wixoaGit/OpenDream/pull/249

**Pull request merged**